internal/processor/parser/app.go

```diff
@@ -170,11 +176,102 @@ func (am *AppParser) Parse(envelope *events.Envelope) ([]metric.MetricPackage, e
 	return metricsPackages, nil
 }
 
+// listApps is meant to replace the function from the go-cfclient and allow fetching apps in parallel tasks
+// The apps returned by this are just missing a reference to the client (unexported property) so don't use the Space() and Summary() methods
+func listApps(c *cfclient.Client, numWorkers int, log *gosteno.Logger) ([]cfclient.App, error) {
+
+	// Query the first page to get the total number of pages.
+	resp, err := queryAppsPage(c, 1)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error requesting apps page 1, skipping cache warmup")
+	}
+	appResources := resp.Resources
+
+	var mutex sync.Mutex
+	var wg sync.WaitGroup
+
+	// Page 1 already fetched
+	pages := resp.Pages - 1
+	// No need for more workers than pages left
+	numWorkers = int(math.Min(float64(numWorkers), float64(pages)))
+	pagesPerWorker := int(math.Ceil(float64(pages) / float64(numWorkers)))
+
+	for i := 0; i < numWorkers; i++ {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			// Offset 2 because no page 0 and page 1 already fetched
+			pageStart := i*pagesPerWorker + 2
+			// Stop at page resp.Pages, which is the last one
+			pageEnd := int(math.Min(float64((i+1)*pagesPerWorker+2), float64(resp.Pages)))
+			resources := queryAppResourcesPageRange(c, pageStart, pageEnd, log)
+			mutex.Lock()
+			appResources = append(appResources, resources...)
+			mutex.Unlock()
+		}(i)
+	}
+
+	wg.Wait()
+
+	apps := []cfclient.App{}
+	for _, app := range appResources {
+		// Taken from https://github.com/cloudfoundry-community/go-cfclient/blob/16c98753d3152f9d80d3c121523536858095a3da/apps.go#L643
+		app.Entity.Guid = app.Meta.Guid
+		app.Entity.CreatedAt = app.Meta.CreatedAt
+		app.Entity.UpdatedAt = app.Meta.UpdatedAt
+		app.Entity.SpaceData.Entity.Guid = app.Entity.SpaceData.Meta.Guid
+		app.Entity.SpaceData.Entity.OrgData.Entity.Guid = app.Entity.SpaceData.Entity.OrgData.Meta.Guid
+		apps = append(apps, app.Entity)
+	}
+	return apps, nil
+}
+
+func queryAppsPage(c *cfclient.Client, pageNb int) (*cfclient.AppResponse, error) {
```

Comment:
Could we rename this to `getApps` or `listApps`. I like get better since it makes a GET request